### ISSUE TITLE
Update to phoenix_live_view 0.9.0

### ISSUE THREAD

**Issue by nelsonic** [dwyl/phoenix-liveview-counter-tutorial#5]
`phoenix_live_view 0.9.0` was released two days ago: https://hex.pm/packages/phoenix_live_view
I've attempted to update to `0.9.0` but it _breaks_:
![image](https://user-images.githubusercontent.com/194400/76304840-f8280480-62bb-11ea-9a41-8581ed7240bf.png)

For reference, this is what it _should_ look like:
![phoenix-liveview-counter-single-windowl](https://user-images.githubusercontent.com/194400/76174551-d6395f80-619f-11ea-8e8d-ab9441d15b6d.gif)

`view-source:http://localhost:4000` suggests that the layout template is not being rendered:
![image](https://user-images.githubusercontent.com/194400/76304980-39b8af80-62bc-11ea-8e04-505fa7c6e374.png)

# Todo

+ [x] Investigate why the layout is not being rendered!

**Comment by nelsonic** [comment]
Reading: https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html#module-live-layouts 👀

**Comment by nelsonic** [comment]
Once #4 is merged we will be able to open an issue on https://github.com/phoenixframework/phoenix_live_view/issues with a full explanation and steps to reproduce!

**Comment by nelsonic** [comment]
Reading: https://github.com/phoenixframework/phoenix_live_view/blob/master/CHANGELOG.md 👀

**Comment by nelsonic** [comment]
Attempted to add `put_root_layout` to `:browser` pipeline in `router.ex` as instructed in the changelog.

Getting:

```sh
== Compilation error in file lib/live_view_counter_web/router.ex ==
** (CompileError) lib/live_view_counter_web/router.ex: undefined function put_root_layout/2
    (elixir 1.10.1) src/elixir_locals.erl:114: anonymous fn/3 in :elixir_locals.ensure_no_undefined_local/3
    (stdlib 3.11.2) erl_eval.erl:680: :erl_eval.do_apply/6
```

There are only 6 Google search results for "phoenix put_root_layout":
https://www.google.com/search?q=phoenix+put_root_layout

<img width="1083" alt="google-search-phoenix-put_root_layout" src="https://user-images.githubusercontent.com/194400/76321470-fc631a80-62d9-11ea-98a8-a395a5fb4497.png">

I've read through all 6 results and [`put_root_layout`](https://github.com/phoenixframework/phoenix/blob/f5ed42cc0442dd2e5f37594a1d2ab6edd393e2cd/lib/phoenix/controller.ex#L571) is a function in the [`controller.ex`](https://github.com/phoenixframework/phoenix/blob/f5ed42cc0442dd2e5f37594a1d2ab6edd393e2cd/lib/phoenix/controller.ex#L571)
![image](https://user-images.githubusercontent.com/194400/76332811-bceffa80-62e8-11ea-8156-4d176463f153.png)

it looks like the Phoenix generator
is about to have a new `--live` option as in:
```sh
mix phx.new --live
```
See: https://github.com/phoenixframework/phoenix/pull/3680

That will _significantly_ reduce the number of setup steps.
I'm going to keep my eyes on the phoenix releases https://github.com/phoenixframework/phoenix/releases
and if a new version is released I will revisit this.

**Comment by nelsonic** [comment]
```
** (exit) an exception was raised:
    ** (ArgumentError) assign @conn not available in eex template.
```

Replaced all instances of `@conn` in `lib/live_view_counter_web/templates/layout/app.html.leex`

Now seeing the following error:
```
[error] GenServer #PID<0.486.0> terminating
** (ArgumentError) :flash is a reserved assign by LiveView and it cannot be set directly. Use the appropriate flash functions instead.
    (phoenix_live_view 0.9.0) lib/phoenix_live_view.ex:1425: Phoenix.LiveView.validate_assign_key!/1
    (phoenix_live_view 0.9.0) lib/phoenix_live_view.ex:1419: anonymous fn/2 in Phoenix.LiveView.assign/2
    (stdlib 3.11.2) maps.erl:232: :maps.fold_1/3
    (live_view_counter 0.9.0) lib/live_view_counter_web/live/counter.ex:25: LiveViewCounterWeb.Counter.handle_info/2
    (phoenix_live_view 0.9.0) lib/phoenix_live_view/channel.ex:113: Phoenix.LiveView.Channel.handle_info/2
    (stdlib 3.11.2) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib 3.11.2) gen_server.erl:711: :gen_server.handle_msg/6
    (stdlib 3.11.2) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: %Phoenix.Socket.Broadcast{event: "inc", payload: %{flash: %{}, live_view_action: nil, live_view_module: LiveViewCounterWeb.Counter, val: 1}, topic: "live"}
State: %{components: {%{}, %{}, 0}, join_ref: "4", serializer: Phoenix.Socket.V2.JSONSerializer, socket: #Phoenix.LiveView.Socket<assigns: %{flash: %{}, live_view_action: nil, live_view_module: LiveViewCounterWeb.Counter, val: 0}, changed: %{}, endpoint: LiveViewCounterWeb.Endpoint, id: "phx-FfsCpTRMyIBxyQBh", parent_pid: nil, root_pid: #PID<0.486.0>, router: LiveViewCounterWeb.Router, view: LiveViewCounterWeb.Counter, ...>, topic: "lv:phx-FfsCpTRMyIBxyQBh", transport_pid: #PID<0.483.0>, uri: %URI{authority: nil, fragment: nil, host: "localhost", path: nil, port: 4000, query: nil, scheme: "http", userinfo: nil}}
```

The line in the code is:
```
 %Phoenix.Socket.Broadcast{
  event: "inc",
  payload: %{
    flash: %{},
    live_view_action: nil,
    live_view_module: LiveViewCounterWeb.Counter,
    val: 1
  },
  topic: "live"
}
```
After a bit of debugging I figured it out! 🙌

**Comment by nelsonic** [comment]
Fixed in #6 :shipit:
